From d99eb300f1f69c8b33f672760660b3a9de97ddaf Mon Sep 17 00:00:00 2001
From: Zhi Wang <zhi.a.wang@intel.com>
Date: Tue, 14 Apr 2015 00:52:21 +0800
Subject: [PATCH 040/403] vgt: set SNPCR as F_PT on BDW.

Signed-off-by: Zhi Wang <zhi.a.wang@intel.com>
---
 drivers/gpu/drm/i915/vgt/handlers.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/i915/vgt/handlers.c b/drivers/gpu/drm/i915/vgt/handlers.c
index 084933f..e112d29 100644
--- a/drivers/gpu/drm/i915/vgt/handlers.c
+++ b/drivers/gpu/drm/i915/vgt/handlers.c
@@ -3334,7 +3334,8 @@ reg_attr_t vgt_base_reg_info[] = {
 {0x7180, 4, F_VIRT, 0, D_ALL, NULL, NULL},
 {0x7408, 4, F_VIRT, 0, D_ALL, NULL, NULL},
 {0x7c00, 4, F_VIRT, 0, D_ALL, NULL, NULL},
-{_REG_SNPCR, 4, F_VIRT, 0, D_ALL, NULL, NULL},
+{_REG_SNPCR, 4, F_VIRT, 0, D_PRE_BDW, NULL, NULL},
+{_REG_SNPCR, 4, F_PT, 0, D_BDW_PLUS, NULL, NULL},
 {_REG_MBCTL, 4, F_VIRT, 0, D_ALL, NULL, NULL},
 {0x911c, 4, F_VIRT, 0, D_ALL, NULL, NULL},
 {0x9120, 4, F_VIRT, 0, D_ALL, NULL, NULL},
@@ -3554,7 +3555,6 @@ reg_attr_t vgt_base_reg_info[] = {
 
 /* UCG */
 {0x8430, 4, F_PT, 0, D_BDW, NULL, NULL},
-{0x900c, 4, F_PT, 0, D_BDW, NULL, NULL},
 
 {0x110000, 4, F_VIRT, 0, D_BDW_PLUS, NULL, NULL},
 };
-- 
1.7.10.4

