From cb57b42dfcfc7cc3ecc7ec192efaa0f96e66895f Mon Sep 17 00:00:00 2001
From: Zhi Wang <zhi.a.wang@intel.com>
Date: Wed, 22 Apr 2015 00:47:41 +0800
Subject: [PATCH 050/403] vgt: let dom0 take control of register 0xfdc

Signed-off-by: Zhi Wang <zhi.a.wang@intel.com>
---
 drivers/gpu/drm/i915/vgt/handlers.c |    5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/drivers/gpu/drm/i915/vgt/handlers.c b/drivers/gpu/drm/i915/vgt/handlers.c
index e112d29..c77dd31 100644
--- a/drivers/gpu/drm/i915/vgt/handlers.c
+++ b/drivers/gpu/drm/i915/vgt/handlers.c
@@ -3532,7 +3532,7 @@ reg_attr_t vgt_base_reg_info[] = {
 {0x913c, 4, F_VIRT, 0, D_BDW, NULL, NULL},
 
 /* WA */
-{0xfdc, 4, F_VIRT, 0, D_BDW, NULL, NULL},
+{0xfdc, 4, F_DOM0, 0, D_BDW, NULL, NULL},
 {0xe4f0, 4, F_RDR, 0, D_BDW, NULL, NULL},
 {0xe4f4, 4, F_RDR, 0, D_BDW, NULL, NULL},
 {0x9430, 4, F_RDR, 0, D_BDW, NULL, NULL},
@@ -3644,9 +3644,6 @@ bool vgt_post_setup_mmio_hooks(struct pgt_device *pdev)
 		reg_update_handlers(pdev, _REG_VCS2_MFX_MODE_BDW, 4,
 				ring_pp_mode_read,
 				ring_pp_mode_write);
-
-		VGT_MMIO_WRITE(pdev, 0xfdc,
-				(1 << 28) | (1 << 24) | (1 << 25) | (1 << 26));
 	}
 
 	return true;
-- 
1.7.10.4

